### ISSUE TITLE
Goflow embedding

### ISSUE THREAD

**Issue by uded** [fieldryand/goflow#86]
I observed that version 2 uses Gin, while v3 is moving to mux, which is fine. But...

In some cases, it would be beneficial to expose middleware that can be used in other frameworks to embed Goflow with Gin or whatever the project choice is. Otherwise, it isn't easy to manage a large deployment of Goflow in complex enterprise environments.

**Comment by fieldryand** [comment]
This is a good point. It would be preferable for the user to have a choice of framework, maybe by exporting the handlers. I will have a think about it. If you have further suggestions, they are more than welcome.

**Comment by shanehughes1990** [comment]
I was just coming across this wanting a dag style workflow but within the same running app much like goflow (very much the core of temporal engine), problem for me is I want to trigger the workflows internally to my app, I wouldn't need the ui just the dag engine itself and looking at the submit route it calls the executor which isn't publicly exposed, would love to see the ability to use the engine without the http server to trigger it

**Comment by fieldryand** [comment]
@shanehughes1990 Good news! This exact thing is coming in the next version. See: https://github.com/fieldryand/goflow/blob/v.3.0.1/goflow.go#L124

**Comment by shanehughes1990** [comment]
That's wonderful I would love to try it out, is it at all a stable branch yet? Enough for a poc?

**Comment by fieldryand** [comment]
Yes, stable enough for a poc. That part of the API will not change.

**Comment by fieldryand** [comment]
You can try `go get github.com/fieldryand/goflow/v3@v3.0.1-alpha.3`
